force_bdss/mco/optimizer_engines/i_optimizer.py

```diff
@@ -0,0 +1,25 @@
+import abc
+import logging
+
+from traits.api import Interface
+
+log = logging.getLogger(__name__)
+
+
+class IOptimizer(Interface):
+    """ The optimizer used by an optimizer engine.
+
+    Extended Summary
+    ----------------
+    An implementing class will be specific to an optimizer library. e.g.
+
+    @provides(IOptimizer)
+    class ScipyOptimizer():
+
+    Any subclass of BaseOptimizerEngine can inherit such a class
+    and then call optimize_function() within its optimize().
+    """
+    @abc.abstractmethod
+    def optimize_function(self, func, x0, bounds):
+        """ Optimize a function, according to the library (scipy/etc).
+        """
```

Comment:
optimize_function() is called from within the engine. I had some more on the docstring but the doc build was complaining.